source/docs/flow_control.md

```diff
@@ -0,0 +1,103 @@
+### Overview
```

Comment:
@htuch obviously fairly rough but I'd appreciate high level comments - is this the sort of detailed walkthrough you were looking for?